packages/open-payments/src/client/requests.ts

```diff
@@ -214,3 +214,34 @@ export const createAxiosInstance = (args: {
 
   return axiosInstance
 }
+
+export type InterceptorFn = (
+  config: InternalAxiosRequestConfig
+) => InternalAxiosRequestConfig | Promise<InternalAxiosRequestConfig>
+
+export const createCustomAxiosInstance = (args: {
+  requestTimeoutMs: number
+  authenticatedRequestInterceptor: InterceptorFn
+}): AxiosInstance => {
+  const axiosInstance = axios.create({
+    headers: {
+      common: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      }
+    },
+    timeout: args.requestTimeoutMs
+  })
+
+  axiosInstance.interceptors.request.use(
+    args.authenticatedRequestInterceptor,
+    undefined,
+    {
+      runWhen: (config: InternalAxiosRequestConfig) =>
+        config.method?.toLowerCase() === 'post' ||
+        !!(config.headers && config.headers['Authorization'])
+    }
+  )
```

Comment:
I think the only thing that would be nice is a small test for this to check that the interceptor was added. I think you can use 
```ts
 createCustomAxiosInstance({
          requestTimeoutMs: 0,
          authenticatedRequestInterceptor: interceptor
        }).interceptors.request['handlers'][0]
```
to get a reference to the interceptor